text/off-chain-workers/indexing.md

```diff
@@ -0,0 +1,60 @@
+# Off-chain Indexing
+
+There are times when on-chain extrinsics need to pass data to the off-chain worker context, such as pass the extrinsic caller account to off-chain worker for further computation.
+
+We can surely pass this piece of data via on-chain storage, but this is costly and it will make the data propagate among the blockchain network. If this is not a piece of critical information that need to have consensus upon the whole network, another way is to pass this data via off-chain indexing.
```

Comment:
Can you think of a real-world motivating example? I guess there must be one if this feature was added to Substrate.

Reply:
@tomusdrw Could you give a real-world example when an on-chain context need to pass data to the off-chain worker context? Or why did you implement this feature in the first place?

Reply:
Like I mentioned earlier, the main use case for `offchain_index` is it's predictable write behavior - for all imported blocks it is guaranteed to run.

Examples in our current codebase include:
1. Writing historic validator sets to offchain db for eqiuvocation reporting purposes (we only keep hash on-chain, so we can verify a proof).
2. Writing leaves & nodes of Merkle Mountain Range, while only keeping peaks (and leave hashes) on-chain.